Quick_Deploy/vLLM/model_repository/vllm/1/model.py

```diff
@@ -0,0 +1,233 @@
+            sampling_params_dict = self.get_sampling_params_dict(request.parameters())
+            sampling_params = SamplingParams(**sampling_params_dict)
+            results_generator = self.llm_engine.generate(
+                str(prompt), sampling_params, request_id
+            )
+
+            # Streaming case
+            async def stream_results() -> AsyncGenerator[bytes, None]:
+                async for request_output in results_generator:
+                    if self._shutdown_event.is_set():
+                        await self.llm_engine.abort(request_id)
+                        yield None
+                    else:
+                        output = self.postprocess(request_output)
+                        triton_output_tensor = pb_utils.Tensor(
+                            "TEXT", np.asarray(output, dtype=self.output_dtype)
+                        )
+                        yield pb_utils.InferenceResponse(
+                            output_tensors=[triton_output_tensor]
+                        )
+
+            if stream:
+                async for response in stream_results():
+                    if response:
+                        response_sender.send(response)
+            # Non-Streaming case
+            else:
+                final_output = None
+                async for request_output in results_generator:
+                    if self._shutdown_event.is_set():
+                        await self.llm_engine.abort(request_id)
+                    else:
+                        final_output = request_output
+                if final_output:
+                    output = self.postprocess(final_output)
+                    triton_output_tensor = pb_utils.Tensor(
+                        "TEXT", np.asarray(output, dtype=self.output_dtype)
+                    )
+                    response = pb_utils.InferenceResponse(
+                        output_tensors=[triton_output_tensor]
+                    )
+                    response_sender.send(response)
+        except Exception as e:
+            self.logger.log_info(f"Error generating stream: {e}")
+            error = pb_utils.TritonError(f"Error generating stream: {e}")
+            triton_output_tensor = pb_utils.Tensor(
+                "TEXT", np.asarray(["N/A"], dtype=self.output_dtype)
+            )
+            response = pb_utils.InferenceResponse(
+                output_tensors=[triton_output_tensor], error=error
+            )
+            response_sender.send(response)
+            raise e
+        finally:
+            response_sender.send(flags=pb_utils.TRITONSERVER_RESPONSE_COMPLETE_FINAL)
+            self.ongoing_request_count -= 1
+
+    def execute(self, requests):
+        """
+        Triton core issues requests to the backend via this method.
+
+        When this method returns, new requests can be issued to the backend.
+        We are pushing all the requests on vllm and let it handle the full traffic.
```

Comment:
You could mention here that if you hold the execute function open, i.e. block it from returning, then with other triton settings, you could effectively timeout queued requests under the condition that either the kv cache was too full or all the slots were full.